pkg/controller/certificate-shim/sync.go

```diff
@@ -160,6 +162,9 @@ func SyncFnFor(
 			if err != nil {
 				return err
 			}
+			if len(extraAnnotations) > 0 {
+				crt.ObjectMeta.Annotations = extraAnnotations
+			}
```

Comment:
suggestion:

1. This seems like it's happening after the call to `Update` above - does this actually get applied to the cert in a cluster?
2. This should probably be appending to `crt.ObjectMeta.Annotations`, not replacing them
3. This code doesn't update the server side apply logic above (lines 143 to 158) so won't work all the time

Reply:
Thanks for review. Indeed, good catch. I'll handle such case